### PR TITLE
Fixes #2 by linking static inline functions and update xdp-tools to v1.4.0

### LINKED ISSUE #2: Missing necessary functions
The following functions in `xsk.h` is not binding in rust
```
xsk_ring_cons__peek

xsk_ring_cons__rx_desc

xsk_ring_prod__reserve

xsk_ring_prod__tx_desc

xsk_ring_cons__comp_addr

xsk_ring_cons__release
```

### PR DESCRIPTION
**Pull request by d0nutptr**
This change does a couple of things to bring the library up-to-date. 

1. This enforces generation of bindings for the previously missing `static inline` functions (`xsk_ring_*`). This fixes #2 
2. Updates bindgen from 0.64.0 to 0.68.1
3. Expands the `.gitignore` since I tend to use JetBrain's based IDEs :) 
4. Makes sure to statically link the associated `libbpf`
5. Updates `xdp-tools` to `v1.4.0` which is the latest tagged release

image

### CODE REVIEW

**Comment by d0nutptr** [comment]
One thing I still need to work on is following https://github.com/rust-lang/rust-bindgen/discussions/2405 and getting thin-lto working so these functions can be inlined (easily) again.

For now, though, this should unblock `libxdp-sys` users who need these functions.

File: .idea/.gitignore
Comment (chenhengqi) [inline .idea/.gitignore:1]: Could we get rid of these `.idea` files ?

**Review by chenhengqi — COMMENTED** [review]
LGTM overall.

Please squash the two commits into one and add your `Signed-off-by` tag.

File: .idea/.gitignore
Comment (d0nutptr) [reply .idea/.gitignore:1]: Yup! I'll update the .gitignore to avoid `**/.idea/` entirely.

**Comment by d0nutptr** [comment]
Should be done now! If it looks good to you, feel free to merge. After this hits main I can publish to crates.io if you'd like. It should already be appropriately versioned for an update.

**Comment by hi-glenn** [comment]
The vesion tag on crate.io may looks better like `libxdp-sys v1.4.0+v1.4.0` .

As the same as   `libbpf-sys v1.2.1+v1.2.0`, a libbpf-sys version tag  and a libbpf version tag.

**Comment by d0nutptr** [comment]
> The vesion tag on crate.io may looks better like `libxdp-sys v1.4.0+v1.4.0` .
> 
> 
> 
> As the same as   `libbpf-sys v1.2.1+v1.2.0`, a libbpf-sys version tag  and a libbpf version tag.

That's a really good idea. I, unfortunately, already pushed the updated library to crates.io. For the next release, though, let's tag it with the libxdp version.

Thanks for the idea!

**Pull request merged**